### DISCUSSION TITLE
API for Lens

### DISCUSSION THREAD

**Discussion by iced-queen** [GitbookIO/community#172] [Feature Requests]
### What's your feature/idea?

I just tried enabling the Lens AI and using it and it's **so good!**
It had a valid answer to all of my questions regarding my project, which is why I'm suggesting an API for it.

I'd be able to implement it on my support bot for my project, which would solve so many questions for our support team!

### What's your desired outcome?

An API where you can make requests with a `text`, `question` or `message` field as examples that would then return the following:
- The answer.
- The related questions:
![img](https://i.gyazo.com/80f8f2f64759a577016a838e0ab282b9.png)
- The `based on` sources:
![img](https://i.gyazo.com/aa58270d2ffbc36e5e73d59e4a060f7e.png)

### What's the impact of this for you?

- [ ] Critical
- [ ] High
- [X] Medium
- [ ] Low

**Comment by SamyPesse** [comment]
Good news, we already have an API for it 🎉 

You can find the documentation at https://developer.gitbook.com/gitbook-api/reference/search

Example with our own documentation:

To get an answer to a question:

```bash
curl  https://api.gitbook.com/v1/spaces/NkEGS7hzeqa35sMXQZ4X/search/ask \
    -H 'content-type: application/json' \
    --data-raw '{"query":"What is GitBook?"}'
```

To list recommended questions:

```bash
curl https://api.gitbook.com/v1/spaces/NkEGS7hzeqa35sMXQZ4X/search/questions
```

**Reply by iced-queen** [reply]
I seem to get an error?

![img](https://i.gyazo.com/48bbd84ee80507e5b37e3812a7bd24ed.png)
```js
let res = await fetch(`https://api.gitbook.com/v1/spaces/${space}/search/ask`, {
    method: "POST",
    headers: {
        Authorization: key,
        "Content-Type": "application/json"
    },
    body: JSON.stringify({
        'query': query
    })
})
console.log(res)
```
